src/App.js

```diff
@@ -12,13 +12,79 @@ const goodsFromServer = [
   'Honey',
   'Jam',
   'Garlic',
+  'Milk',
+  'Bear',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  }
+
+  addToSelected = (product) => {
+    if (!this.state.selectedGoods.includes(product)) {
+      this.setState(state => ({
+        selectedGoods: [...state.selectedGoods, product],
+      }));
+    } else {
+      this.setState(state => ({
+        selectedGoods: state.selectedGoods.filter(good => good !== product),
+      }));
+    }
+  }
+
+  clearSelectedGoods = () => {
+    this.setState({ selectedGoods: [] });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1 className="SelectedGoods">
+          Selected goods:
+          {' '}
+          {
+            this.state.selectedGoods.map((good, index, array) => {
+              if (array.length > 1 && index === array.length - 2) {
+                return `${good} and `;
+              }
+
+              return `${good} `;
+            })
+          }
+          <button
+            type="button"
+            onClick={this.clearSelectedGoods}
+          >
+            X
+          </button>
```

Comment:
Aslo here, you need to remove your button when no goods selected.
12. The clear button should not be visible if there is no selected good

Reply:
add condition if no selected goods then don't show this button